crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -45,9 +45,9 @@ impl Parse for AllTuples {
 #[proc_macro]
 pub fn all_tuples(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as AllTuples);
-    let len = input.end - input.start;
+    let len = (input.start..=input.end).count();
```

Comment:
👍 I was going to suggest this, but didn't know what the syntax would be.

I can't remember if this is efficiently implemented - it might be one of the range footguns.

Reply:
It should be a small footgun, as it's not called with a long range and it's at compile time